### PR TITLE
Update to Python3 for CPython code

### PR DESCRIPTION
**Pull request by jjwilke**

### CODE REVIEW

**Comment by sknigh** [comment]
I was not able to build this on Ubuntu with system python, as it would always select `/usr/bin/python` (python 2) instead of `/usr/bin/python3`.

Loading a python3 module from spack worked because it prepends the search path, and the `bin` directory's `python` actually links to python3.

`make check` passed, but `make installcheck` printed the following error
```
chmod -R u+w skeletons # required for make distcheck (makes files r/o)
cd skeletons/sst_component_example; PATH=/home/sknigh/code/github/sst-macro/build/install/bin:$PATH make
make[3]: Entering directory '/home/sknigh/code/github/sst-macro/build/tests/skeletons/sst_component_example'
sst++ -I.  -fPIC --sst-component -c component.cc -o component.o
/home/sknigh/code/github/sst-macro/build/install/bin/sst++: 2: /home/sknigh/code/github/sst-macro/build/install/bin/sst++: __license__: not found
from: can't read /var/mail/sstcclib
import-im6.q16: unable to open X server `' @ error/import.c/ImportImageCommand/358.
/home/sknigh/code/github/sst-macro/build/install/bin/sst++: 48: /home/sknigh/code/github/sst-macro/build/install/bin/sst++: Syntax error: "(" unexpected
Makefile:30: recipe for target 'component.o' failed
```

**Comment by jjwilke** [comment]
@sknigh can you post your configure line?

File: sstmac/sst_core/integrated_core.cc [hunk 1]
```diff
@@ -127,6 +133,7 @@ py_extract_params(PyObject* dict, sprockit::SimParameters::ptr params)
     Py_DECREF(pair);
   }
   Py_DECREF(items);
+  return true;
```
Comment (calewis) [inline sstmac/sst_core/integrated_core.cc:136]: Function always returns true?

File: sstmac/sst_core/integrated_core.cc [hunk 2]
```diff
@@ -112,9 +112,15 @@ py_extract_params(PyObject* dict, sprockit::SimParameters::ptr params)
...
+    bool isMapping = PyMapping_Check(val);
+#if PY_MAJOR_VERSION >= 3
+    bool isString = PyUnicode_Check(val);
+#else
+    bool isString = false; //ignore, not needed
+#endif
+    if (isMapping && !isString){
       sprockit::SimParameters::ptr sub_params = params->getOptionalNamespace(key_c_str);
-      sstmac::py_extract_params(val, sub_params);
+      isMapping = sstmac::py_extract_params(val, sub_params);
```
Comment (calewis) [inline sstmac/sst_core/integrated_core.cc:123]: I don't see any looping this looks like a dead write to `isMapping`

File: sstmac/sst_core/integrated_core.cc [hunk 3]
```diff
@@ -100,7 +100,7 @@ int_vector_from_py_array(PyObject* tuple, std::vector<int>& vec)
   }
 }
 
-void
+bool
```
Comment (calewis) [inline sstmac/sst_core/integrated_core.cc:103]: Aside from the dead write below I don't think this return value is ever actually used?

File: sstmac/sst_core/integrated_core.cc (hunk 1 above)
Comment (jjwilke) [reply sstmac/sst_core/integrated_core.cc]: Removed.

File: sstmac/sst_core/integrated_core.cc (hunk 2 above)
Comment (jjwilke) [reply sstmac/sst_core/integrated_core.cc]: removed

File: sstmac/sst_core/integrated_core.cc (hunk 3 above)
Comment (jjwilke) [reply sstmac/sst_core/integrated_core.cc]: removed

**Comment by jjwilke** [comment]
@sknigh: I think my most recent push should fix the ubuntu issues
@calewis: I reverted the code. The bool return is not needed.

File: sstmac/sst_core/python_topology.cc [hunk 4]
```diff
@@ -210,7 +231,7 @@ sys_convert_to_list(const std::vector<sstmac::hw::Topology::InjectionPort>& port
 static PyObject*
 sys_get_injection_connections(SystemPy_t* self, PyObject* swIdx)
 {
-  int sid = PyInt_AsLong(swIdx);
+  int sid = ConvertToCppLong(swIdx);
```
Comment (calewis) [inline sstmac/sst_core/python_topology.cc:234]: `int` isn't a `long`.

**Pull request merged**